.ci/benchmarks.groovy

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'linux && immutable && obs11' }
+  environment {
+    REPO = 'apm-server'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    BRANCH_NAME_LOWER_CASE = "${env.BRANCH_NAME.toLowerCase().replaceAll('[^a-z0-9-]', '-')}"
+    AWS_ACCOUNT_SECRET = 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+    GCP_ACCOUNT_SECRET = 'secret/observability-team/ci/elastic-observability-account-auth'
+    EC_KEY_SECRET = 'secret/observability-team/ci/elastic-cloud/observability-pro'
+    TERRAFORM_VERSION = '1.1.9'
+
+    CREATED_DATE = "${new Date().getTime()}"
+    JOB_GCS_BUCKET_STASH = 'apm-ci-temp'
+    JOB_GCS_CREDENTIALS = 'apm-ci-gcs-plugin'
+  }
+
+  options {
+    timeout(time: 2, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+
+  stages {
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", shallow: false)
+      }
+    }
+
+    stage('Benchmarks') {
+      options { skipDefaultCheckout() }
+      steps {
+        dir ("${BASE_DIR}") {
+          withGoEnv() {
+            dir("testing/benchmark") {
+              withTestClusterEnv {
+                  sh(label: 'Build apmbench', script: 'make apmbench $SSH_KEY terraform.tfvars')
```

Comment:
What's the purpose of using the `$SSH_KEY` make goal? I cannot find any references in https://github.com/elastic/apm-server/blob/main/testing/benchmark/Makefile

OTOH, if it's needed, I'd say `SSH_KEY` to be declared in the `environment` section either in the top level environment definition or create a new one in the `stage('Benchmarks')`, since the definition in 'L75` with the `withEnv` is not easy to debug

Reply:
`$SSH_KEY` is path to generated ssh key to connect to the benchmark executor machine. 
Will get rid of `withEnv`